v19.1/cost-based-optimizer.md

```diff
@@ -328,6 +331,236 @@ $ cockroach sql --insecure --host=localhost --port=26259 --database=test -e 'EXP
 
 You'll need to make changes to the above configuration to reflect your [production environment](recommended-production-settings.html), but the concepts will be the same.
 
+### Example - leaseholder preferences
+
+If you provide [leaseholder preferences](configure-replication-zones.html#lease_preferences) in addition to replication zone constraints, the optimizer will attempt to take your leaseholder preferences into account as well when selecting an index for your query. There are several factors to keep in mind:
+
+- Zone constraints are always respected (hard constraint), whereas lease preferences are taken into account as "additional information" -- as long as they don't contradict the zone constraints.
+
+- The optimizer doesn't consider the real-time location of leaseholders when selecting an index; it is pattern matching on the text values passed in the configuration (e.g., the [`ALTER INDEX`](alter-index.html) statements shown below). For the same reason, the optimizer only matches against the first locality in your `lease_preferences` array.
+
+- In some cases, it may not be possible for the optimizer to use an index that satisfies your leaseholder preferences, since [leaseholders can move](architecture/replication-layer.html#leaseholder-rebalancing)  to different nodes. Therefore, you should only use this feature if you’re confident you know where the leaseholders will end up based on your cluster's usage patterns. We recommend thoroughly testing your configuration to ensure the optimizer is selecting the index(es) you expect.
```

Comment:
It's always "possible" for the optimizer to use any index. It just may be that it decides to use an index based on leaseholder preferences, even though that index has moved to a different node/region, thus causing slower performance than expected.

Reply:
Rewrote to use this phrasing/info instead - PTAL.